source/templates/redirect.html

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Redirect</title>
+</head>
+<body>
+<script>
+    const DEFAULT_PAGE = "-";
+    // The line below is injected; all other lines are constant
+    {{ redirects_object }}
+    // The line above is injected; all other lines are constant
+    const url = new URL(window.location.href);
+    let redirectKey = DEFAULT_PAGE;
+    if (url.hash !== "" && url.hash !== "#") {
+        redirectKey = url.hash.substring(1);
```

Comment:
We'll want to add the input validation here too.

Reply:
I've added fragment validation to this code.